test/conftest.py

```diff
@@ -21,496 +21,99 @@
-        self.testwindows.remove(proc)
-
-        @Retry(ignore_exceptions=(ValueError,))
-        def success():
-            if len(self.c.windows()) < start:
-                return True
-            raise ValueError('window is still in client list!')
-
-        if not success():
-            raise AssertionError("Window could not be killed...")
-
-    def test_window(self, name, type="normal"):
-        """
-        Windows created with this method must have their process killed explicitly, no
-        matter what type they are.
-        """
-        return self._spawn_script("window.py", self.display, name, type)
-
-    def test_dialog(self, name="dialog"):
-        return self.test_window(name, "dialog")
-
-    def test_notification(self, name="notification"):
-        return self.test_window(name, "notification")
-
-    def test_xclock(self):
-        path = shutil.which("xclock")
-        return self._spawn_window(path)
-
-    def test_xeyes(self):
-        path = shutil.which("xeyes")
-        return self._spawn_window(path)
-
-    def test_xcalc(self):
-        path = shutil.which("xcalc")
-        return self._spawn_window(path)
-
-    def groupconsistency(self):
-        groups = self.c.groups()
-        screens = self.c.screens()
-        seen = set()
-        for g in groups.values():
-            scrn = g["screen"]
-            if scrn is not None:
-                if scrn in seen:
-                    raise AssertionError(
-                        "Screen referenced from more than one group.")
-                seen.add(scrn)
-                assert screens[scrn]["group"] == g["name"]
-        assert len(seen) == len(screens), "Not all screens \
-        had an attached group."
+def pytest_cmdline_main(config):
+    if not config.option.backend:
+        config.option.backend = ["x11"]
 
+    ignore = config.option.ignore or []
+    if "wayland" not in config.option.backend:
+        ignore.append("test/backend/wayland")
+    if "x11" not in config.option.backend:
+        ignore.append("test/backend/x11")
+    config.option.ignore = ignore
 
-@pytest.fixture(scope="session")
-def xvfb():
-    with xcffib.testing.XvfbTest():
-        display = os.environ["DISPLAY"]
-        if not can_connect_x11(display):
-            raise OSError("Xvfb did not come up")
 
-        yield
+def pytest_generate_tests(metafunc):
+    if "backend" in metafunc.fixturenames:
+        backends = metafunc.config.option.backend
+        metafunc.parametrize("backend_name", backends)
 
 
-@pytest.fixture(scope="session")
-def display(xvfb):  # noqa: F841
-    return os.environ["DISPLAY"]
+@pytest.fixture(scope="session", params=[1])
+def outputs(request):
+    return request.param
+
+
+dualmonitor = pytest.mark.parametrize("outputs", [2], indirect=True)
+multimonitor = pytest.mark.parametrize("outputs", [1, 2], indirect=True)
 
 
 @pytest.fixture(scope="session")
-def xephyr(request, xvfb):  # noqa: F841
+def xephyr(request, outputs):  # noqa: F841
```

Comment:
I think we can drop the noqa comment here and in the next fixture as well.

Reply:
:+1: